trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogInjector.java

```diff
@@ -127,18 +127,34 @@ public void init(ConnectorContext context) {
       }
 
       // 3. Retrieve the catalogManager object.
-      MetadataProvider metadataProvider = context.getMetadataProvider();
-
-      Object metadata = getFiledObject(metadataProvider, "metadata");
-      Object metadataManager = metadata;
-      if (isClassObject(metadata, "TracingMetadata")) {
-        metadataManager = getFiledObject(metadata, "delegate");
+      Object catalogManager = null;
+
+      try {
+        // Try to retrieve catalogManager from static filed of
+        // "io.trino.server.Server.trinoCatalogManager"
+        // If the Trino is build with the patch
+        // https://github.com/datastrato/Trino/commit/93b91791f9148c1885c5fac4d59cd5960feec685
```

Comment:
This is not an open- source project, and users can not review the code.

Reply:
Furthermore, please provide more details about why we need the changes and the usage cases.

Reply:
If this is a in house fix, maybe we don't need this PR for Gravitino?

Reply:
This is to facilitate our deployment. If we provide users with the internal version of Trino, they can directly use the released Trino connector without the need to modify and recompile it.

Reply:
What's the meaning of `they can directly use the released Trino connector without the need to modify and recompile it`, should users need to modify the Trino connector if we use the community version?

Anyway, I don't think we should expose our internal version of Trino to users for any purpose.